haystack/document_store/elasticsearch.py

```diff
@@ -121,6 +128,46 @@ def __init__(
         self.update_existing_documents = update_existing_documents
         self.refresh_type = refresh_type
 
+    def _init_elastic_client(self,
+                             host: Union[str, List[str]],
+                             port: Union[int, List[int]],
+                             username: str,
+                             password: str,
+                             api_key_id: Optional[str],
+                             api_key: Optional[str],
+                             scheme: str,
+                             ca_certs: Optional[str],
+                             verify_certs: bool,
+                             timeout: int) -> Elasticsearch:
+        # Create list of host(s) + port(s) to allow direct client connections to multiple elasticsearch nodes
+        if isinstance(host, list):
+            if isinstance(port, list):
+                if not len(port) == len(host):
+                    raise ValueError("Length of list `host` must match length of list `port`")
+                hosts = [{"host":h, "port":p} for h, p in zip(host,port)]
+            else:
+                hosts = [{"host": h, "port": port} for h in host]
+        else:
+            hosts = [{"host": host, "port": port}]
+
+        if (api_key or api_key_id) and not(api_key and api_key_id):
+            raise ValueError("You must provide either both or none of `api_key_id` and `api_key`")
+
+        # Init client based on authentication mode
+        try:
+            if api_key:
+                # api key authentication
+                return Elasticsearch(hosts=hosts, http_auth=(username, password),
+                                            scheme=scheme, ca_certs=ca_certs, verify_certs=verify_certs,
+                                            timeout=timeout)
+            else:
+                # standard http_auth
+                return Elasticsearch(hosts=hosts, api_key=(api_key_id, api_key),
+                                            scheme=scheme, ca_certs=ca_certs, verify_certs=verify_certs, timeout=timeout)
+        except Exception as e:
```

Comment:
Last time I tested the Elasticsearch client returned only warnings, but no exceptions in case of connection failure. If that's still the case, this `except` block will never get executed. We could validate the behavior by adding a test case for connecting to an invalid host.

Reply:
Changed it to a manual `ping()` call to test the connection